### PR TITLE
refactor: split cli core e2e tests

### PR DESCRIPTION
**Pull request by xeno097**
### Description

Splits cli core e2e tests to reduce test execution and reduces constants duplicity across test files

### Backward compatibility

- Yes

### Testing

- Manual/e2e

### CODE REVIEW

File: typescript/cli/src/tests/core-apply.e2e-test.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+} from './commands/helpers.js';
+
+describe('hyperlane core apply e2e tests', async function () {
+  this.timeout(2 * DEFAULT_E2E_TEST_TIMEOUT);
+
+  let signer: Signer;
+  let initialOwnerAddress: Address;
+
+  before(async () => {
+    const chainMetadata: any = readYamlOrJson(
```
Comment (jmrossy) [inline typescript/cli/src/tests/core-apply.e2e-test.ts:35]: why any?

File: typescript/cli/src/tests/core-deploy.e2e-test.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+const EXAMPLES_PATH = './examples';
+const CORE_CONFIG_PATH = `${EXAMPLES_PATH}/core-config.yaml`;
+
+const TEMP_PATH = '/tmp'; // /temp gets removed at the end of all-test.sh
+const CORE_READ_CONFIG_PATH = `${TEMP_PATH}/${CHAIN_NAME}/core-config-read.yaml`;
+
+const TEST_TIMEOUT = 100_000; // Long timeout since these tests can take a while
+describe('hyperlane core deploy e2e tests', async function () {
+  this.timeout(TEST_TIMEOUT);
```
Comment (jmrossy) [inline typescript/cli/src/tests/core-deploy.e2e-test.ts:26]: Should this use DEFAULT_E2E_TEST_TIMEOUT

File: typescript/cli/src/tests/core-deploy.e2e-test.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import { hyperlaneCoreDeploy, readCoreConfig } from './commands/core.js';
+import { ANVIL_KEY, REGISTRY_PATH } from './commands/helpers.js';
+
+const CHAIN_NAME = 'anvil2';
+
+const EXAMPLES_PATH = './examples';
+const CORE_CONFIG_PATH = `${EXAMPLES_PATH}/core-config.yaml`;
+
+const TEMP_PATH = '/tmp'; // /temp gets removed at the end of all-test.sh
+const CORE_READ_CONFIG_PATH = `${TEMP_PATH}/${CHAIN_NAME}/core-config-read.yaml`;
```
Comment (jmrossy) [inline typescript/cli/src/tests/core-deploy.e2e-test.ts:22]: These can also be de-duped right?

**Review by jmrossy — COMMENTED** [review]
Mostly looks good

File: typescript/cli/src/tests/core-deploy.e2e-test.ts (hunk 3 above)
Comment (xeno097) [reply typescript/cli/src/tests/core-deploy.e2e-test.ts:22]: you are right missed these, thanks!

File: typescript/cli/src/tests/core-apply.e2e-test.ts (hunk 1 above)
Comment (xeno097) [reply typescript/cli/src/tests/core-apply.e2e-test.ts]: Just copy pasted from the old test, fixed it to use the `ChainMetadata` type!

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -126,7 +126,9 @@ jobs:
       fail-fast: false
       matrix:
         test:
-          - core
+          - core-apply
+          - core-deploy
+          - core-read
           - relay
```
Comment (paulbalaji) [inline .github/workflows/test.yml:132]: it's beautiful

File: .changeset/four-pumpkins-glow.md
Comment (paulbalaji) [inline .changeset/four-pumpkins-glow.md:1]: do we need changesets for tests?

**Review by paulbalaji — APPROVED** [review]
this is looking great, thank you

File: .changeset/four-pumpkins-glow.md
Comment (xeno097) [reply .changeset/four-pumpkins-glow.md:1]: Good question do we?

Comment (paulbalaji) [reply .changeset/four-pumpkins-glow.md:1]: I personally don't think so 🤷

Comment (jmrossy) [reply .changeset/four-pumpkins-glow.md:1]: Nah

**Review by jmrossy — APPROVED** [review]

**Pull request merged**